zerver/views/webhooks/trello/board_actions.py

```diff
@@ -0,0 +1,107 @@
+    REMOVE_MEMBER: 'removed {member_name} from {board_url_template}',
+    ADD_MEMBER: 'added {member_name} to {board_url_template}',
+    CREATE_LIST: 'added {list_name} list to {board_url_template}',
+    CHANGE_NAME: 'renamed the board from {old_name} to {board_url_template}'
+}
+
+def process_board_action(payload, action_type):
+    # type: (Dict[str, Any], str) -> Tuple[str, str]
+    action_type = get_proper_action(payload, action_type)
+    return get_subject(payload), get_body(payload, action_type)
+
+def get_proper_action(payload, action_type):
+    # type: (Dict[str, Any], str) -> str
+    if action_type == 'updateBoard':
+        data = get_action_data(payload)
+        if data.get('old').get('name'):
+            return CHANGE_NAME
+        raise UnknownUpdateBoardAction()
+    return action_type
+
+def get_subject(payload):
+    # # type: (Dict[str, Any]) -> str
+    data = {
+        'board_name': get_action_data(payload).get('board').get('name')
+    }
+    return TRELLO_SUBJECT_TEMPLATE.format(**data)
+
+def get_body(payload, action_type):
+    # type: (Dict[str, Any], str) -> str
+    message_body = ACTIONS_TO_FILL_BODY_MAPPER[action_type](payload, action_type)
+    creator = payload.get('action').get('memberCreator').get('fullName')
+    return TRELLO_MESSAGE_TEMPLATE.format(full_name=creator, rest=message_body)
+
+def get_managed_member_body(payload, action_type):
+    # type: (Dict[str, Any], str) -> str
+    data = {
+        'member_name': payload.get('action').get('member').get('fullName'),
+    }
+    return fill_appropriate_message_content(payload, action_type, data)
+
+def get_create_list_body(payload, action_type):
+    # type: (Dict[str, Any], str) -> str
+    data = {
+        'list_name': get_action_data(payload).get('list').get('name'),
+    }
+    return fill_appropriate_message_content(payload, action_type, data)
+
+def get_change_name_body(payload, action_type):
+    # type: (Dict[str, Any], str) -> str
+    data = {
+        'old_name': get_action_data(payload).get('old').get('name'),
+    }
+    return fill_appropriate_message_content(payload, action_type, data)
+
+
+def fill_appropriate_message_content(payload, action_type, data=None):
+    # type: (Dict[str, Any], str, Any) -> str
+    data = {} if data is None else data
+    data['board_url_template'] = data.get('board_url_template', get_filled_board_url_template(payload))
+    message_body = get_message_body(action_type)
+    return message_body.format(**data)
+
+def get_filled_board_url_template(payload):
+    # type: (Dict[str, Any]) -> str
+    return TRELLO_BOARD_URL_TEMPLATE.format(board_name=get_board_name(payload), board_url=get_board_url(payload))
+
+def get_board_name(payload):
+    # type: (Dict[str, Any]) -> str
+    return get_action_data(payload).get('board').get('name')
+
+def get_board_url(payload):
+    # type: (Dict[str, Any]) -> str
+    return 'https://trello.com/b/{}'.format(get_action_data(payload).get('board').get('shortLink'))
```

Comment:
This should be `u'https://trello.com/b/{}'`. We type all URLs as `text_type` since URLs can contain non-ASCII characters. This one will most likely not, but URLs can contain non-ASCII characters in general, so it's better to consistently give the same type to all URLs everywhere in Zulip.